hazelcast/src/main/java/com/hazelcast/internal/server/tcp/TcpServerConnectionManager.java

```diff
@@ -344,4 +354,22 @@ public void provideDynamicMetrics(MetricDescriptor descriptor, MetricsCollection
             context.collect(rootDescriptor.copy(), TCP_METRIC_TEXT_COUNT, MANDATORY, COUNT, textCount);
         }
     }
+
+    @Override
+    public boolean blockOnConnect(Address address, long millis, int streamId) throws InterruptedException {
+        Plane plane = getPlane(streamId);
+        try {
+            Future<Void> future = plane.connectionsInProgress.get(address);
+            if (future != null) {
+                future.get(millis, TimeUnit.MILLISECONDS);
+                return future.isDone();
+            } else {
+                // connection-in-progress has come and gone,
+                // so we are not timed out here
+                return true;
+            }
+        } catch (ExecutionException | TimeoutException ex) {
+            throw ExceptionUtil.rethrow(ex);
```

Comment:
No need to rethrow here, just ignore any exceptions. The `ConnectTask` itself catches all throwables and error handling happens in other parts of the code. You can also catch `CancellationException` and ignore it here.

Reply:
Hiding exceptions is a big code smell. If the code is ever changed inside ConnectTask, this will fail correctly.

Reply:
agreed, leave the `rethrow` logic as is.